### PR TITLE
Fix tests with latest go cmp

### LINKED ISSUE #155: TestDeepEqualFailure fails w/ go 1.12.2
gotest.tools 2.3.0 w/ Go 1.12.2 on Fedora Rawhide, x86_64:

```
Testing    in: /builddir/build/BUILD/gotest.tools-2.3.0/_build/src
         PATH: /builddir/build/BUILD/gotest.tools-2.3.0/_build/bin:/builddir/.local/bin:/builddir/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin
       GOPATH: /builddir/build/BUILD/gotest.tools-2.3.0/_build:/usr/share/gocode
  GO111MODULE: off
      command: go test -buildmode pie -compiler gc -ldflags "-X gotest.tools/version=2.3.0 -extldflags '-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '"
      testing: gotest.tools
gotest.tools/assert
--- FAIL: TestDeepEqualFailure (0.00s)
    assert_test.go:305: should have failure message "assertion failed: \n--- actual\n+++ expected\n{assert.stub}.b:\n\t-: 1\n\t+: 2\n", got "assertion failed: \n--- actual\n+++ expected\n\u00a0\u00a0assert.stub{\n\u00a0\u00a0\ta: \"ok\",\n-\u00a0\tb: 1,\n+\u00a0\tb: 2,\n\u00a0\u00a0}\n"
FAIL
exit status 1
FAIL	gotest.tools/assert	0.028s
```

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #155

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like tests are failing (on older Go versions?) @dnephin 
Wondering if those versions should still be tested though, because Go 1.10 and older reached EOL

**Comment by dnephin** [comment]
Ya, I'd like to find a way to test it that works with all versions

File: assert/cmp/compare_test.go [hunk 1]
```diff
@@ -15,16 +15,7 @@ import (
...
---- result
-+++ exp
-{[]string}[0]:
-	-: "a"
-	+: "b"
-{[]string}[1]:
-	-: "b"
-	+: "a"
-`
+	expected := "\n--- result\n+++ exp\n  []string{\n- \t\"a\",\n+ \t\"b\",\n- \t\"b\",\n+ \t\"a\",\n  }\n"
```
Comment (cbandy) [inline assert/cmp/compare_test.go]: Perhaps use regex for the whitespace: `[\pZ\s]`

**Comment by dnephin** [comment]
Alright, tests seem to be passing now

**Pull request merged**